src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.PackTool.targets

```diff
@@ -59,6 +59,10 @@ Copyright (c) .NET Foundation. All rights reserved.
         <PackagePath>%(_ResolvedFileToPublishWithPackagePath.PackagePath)</PackagePath>
       </TfmSpecificPackageFile>
     </ItemGroup>
+
+    <!-- Enable warning for trying to use PackRelease with a solution if env-var is not set.-->
+    <NETSdkWarning Condition="'$(PackRelease)' != '' and '$(SolutionExt)' == '.sln' and '$(ENABLE_P_RELEASE_SLN)' == ''"
```

Comment:
This is in the `PackTool` target, which I believe is only run when packing projects where `PackAsTool` is true.  So this won't apply to most cases of `dotnet pack`.

Reply:
(All) of the `pack` targets seem to be in `NuGet.build.tasks`. So I'd need to raise a PR in NuGet. I'll wait for your response on the env name above.